frontend/src/components/register/register.js

```diff
@@ -1,181 +1,221 @@
-												fullWidth
-											/>
+										<Grid item xs='6'>
+											<Field 
+											name='lastName' 
+											id='last_name' 
+											label='Last' 
+											type='text'
+											variant='outlined' 
+											required
+											component={inputField} />
 										</Grid>
 									</Grid>
-									<Grid item>
-									<button type="submit" className={classes.Submit} onClick={this.handleSubmit}>SIGN UP</button>
+									<Grid item >
+											<Field 
+											name='email' 
+											id='email'
+											label='E-mail'
+											type='email'
+											variant='outlined'
+											required
+											component={inputField} />
+									</Grid>
+									<Grid item >
+											<Field 
+											name='password' 
+											id='password'
+											label='Password'
+											type='password'
+											variant='outlined'
+											required
+											component={inputField} />
 									</Grid>
+									<Grid container item direction='row' spacing='16'>
+										<Grid item xs='8'>
+											<Field 
+											name='school' 
+											id='school'
+											label='School' 
+											type='text'
+											variant='outlined'
+											required
+											component={selectField}>
+												{schools ? (
+													schools.map((school) => (
+													<MenuItem value={school.id}>{school.code}</MenuItem>
+												))) : (
+													<p>loading</p>
+												)}
+											</Field>
+										</Grid>
+									<Grid item xs='4'>
+											<Field 
+											name='major' 
+											id='major'
+											label='Major' 
+											type='text'
+											variant='outlined'
+											component={inputField} />
+									</Grid>
+								</Grid>
+								<Grid item>
+								<button 
+								type="submit"
+								disabled={submitting || pristine}
+								className={classes.Submit} >SIGN UP</button>
 								</Grid>
 							</Grid>
-							<Snackbar
-								open={openAlert}
-								onClose={this.handleAlertClose}
-								message={<span>Incomplete Form</span>}
-							/>
+						</Grid>
+{/* 							<Snackbar
+							open={openAlert}
+							onClose={this.handleAlertClose}
+							message={<span>Incomplete Form</span>}
+						/> */}
 						</form>
-						<div>
-							<span>Already have a StudY Account?</span>&ensp;
-							<a href="/login" className={classes.Signin}>SIGN IN</a>
-						</div>
-                    </div>
+					<div>
+						<span>Already have a StudY Account?</span>&ensp;
+						<a href="/login" className={classes.Signin}>SIGN IN</a>
+					</div>
 				</div>
 			</div>
 		</div>
-		)
-	}
+	</div>
+	)
 }
 
-const mapStateToProps = state => {};
+Register.propTypes = {
+	schools: propTypes.array,
+	applicant: propTypes.object,
+}
+
+Register.defaultProps = {
+	applicant: {},
+	schools: [],
+}
 
-export default connect(mapStateToProps,{ registerUser })(Register);
+export default reduxForm({
+	form: 'registerForm',
+	validate,
```

Comment:
is there a reason you're doing async validation in `Register`? it seems that change/blur events set `asyncValidating` so redux-form expects a promise and your `validate` is synchronous. This causes the form to become laggy/unresponsive. If you're not fetching data, try this :

```suggestion
	validate,
	asyncBlurFields: [],
```

![peek 2018-10-29 18-25](https://user-images.githubusercontent.com/9379035/47685910-d9d1d500-dbae-11e8-8c8f-91f35ab51c6a.gif)

Reply:
ahh I was trying to figure this out. Guess I missed that part of the docs :/